### ISSUE TITLE
`must_get_agent_activity` can't query the agent's own chain

### ISSUE THREAD

**Issue by ThetaSinner** [holochain/holochain#4200]
https://chat.holochain.org/holo/pl/8eu5x4aagfbn3pbinj4ptsn1ko

**Comment by jost-s** [comment]
query is not run against the authored database

**Comment by jost-s** [comment]
In the cascade, we're doing the expected "try this query on each database" pattern -> https://github.com/holochain/holochain/blob/develop/crates/holochain_cascade/src/lib.rs#L968

Unfortunately, following this down we find the bounds lookup -> https://github.com/holochain/holochain/blob/develop/crates/holochain_cascade/src/authority/get_agent_activity_query/must_get_agent_activity.rs#L87 and the SQL that it uses -> https://github.com/holochain/holochain/blob/develop/crates/holochain_sqlite/src/sql/cell/agent_activity/action_hash_to_seq.sql

That SQL grabs everything back and it looks from some quick debugging that everything will be fine. However, rather than failing cleanly there, we continue on and use another query -> https://github.com/holochain/holochain/blob/develop/crates/holochain_sqlite/src/sql/cell/agent_activity/must_get_agent_activity.sql

Now we're filtering out anything which doesn't have a `when_integrated` set. That's never the case in the authored database. It's only ever used in the DHT database. 

So basically, unless you have content in the DHT or cache databases, you'd fudged at that point. You can't get your own activity back out of the authored database to answer a `must_get_agent_activity`.

You might be able to fall back to the network an inefficiently cache your own ops... Except the authority check was failing there due to the first issue. I've now "fixed" that... So we will go to the network to retrieve out own agent activity on a sharded network where we aren't an authority for ourselves, which is really quite wrong.

Question:
- What is a reproduction of this problem? When a zome call writes data to the source chain, it's also copied to the DHT database and integrated. So a must_get_agent_activity query should be able to find the recently written data.

**Comment by jost-s** [comment]
It turns out this issue does not exist. [Ops written to the authored database during the call zome workflow are set to integrated](https://github.com/holochain/holochain/blob/develop/crates/holochain_state/src/mutations.rs#L288).
I've written a test to prove that it works, and commented the above line out, which made the test fail. Equally removing the `WHERE when_integrated IS NOT NULL` clause from the corresponding SQL lets the test pass again.
